### PR TITLE
Test Parameter

### PR DESCRIPTION
**Pull request by ramz7**

### CODE REVIEW

File: tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/repositories/TestParameterRepository.java [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
-public interface TestParameterRepository {
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import com.tokyo.supermix.data.entities.TestParameter;
 
+@Repository
+public interface TestParameterRepository extends JpaRepository<TestParameter, Long> {
+	List<TestParameter> findByTestId(Long testid);
```
Comment (Mathuragini) [inline tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/repositories/TestParameterRepository.java]: use camel case testId

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/TestParameterController.java [hunk 2]
```diff
@@ -1,9 +1,124 @@
...
+    }
+    testParameterService
+        .saveTestParameter(mapper.map(testParameterRequestDto, TestParameter.class));
+    return new ResponseEntity<>(
+        new BasicResponse<>(RestApiResponseStatus.OK, Constants.ADD_TEST_PARAMETER_SUCCESS),
+        HttpStatus.OK);
+  }
+
+  @GetMapping(value = EndpointURI.TEST_PARAMETERS)
+  public ResponseEntity<Object> getAllRawMaterials() {
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/TestParameterController.java]: change the method name: getAllTestParameters()

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/TestParameterController.java [hunk 3]
```diff
@@ -1,9 +1,124 @@
...
+    return new ResponseEntity<>(new ContentResponse<>(Constants.TEST_PARAMETERS,
+        mapper.map(testParameterService.getAllTestParameters(), TestParameterResponseDto.class),
+        RestApiResponseStatus.OK), null, HttpStatus.OK);
+  }
+
+  @GetMapping(value = EndpointURI.TEST_PARAMETER_BY_ID)
+  public ResponseEntity<Object> getTestParameterByTestParameterId(@PathVariable("id") Long id) {
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/TestParameterController.java]: @PathVaraible(id),
no need to add - (id)

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/TestParameterController.java [hunk 4]
```diff
@@ -1,9 +1,124 @@
...
+          RestApiResponseStatus.OK), HttpStatus.OK);
+    }
+    return new ResponseEntity<>(new ValidationFailureResponse(Constants.TEST_PARAMETER_ID,
+        validationFailureStatusCodes.getTestParameterNotExist()), HttpStatus.BAD_REQUEST);
+  }
+
+  @DeleteMapping(EndpointURI.TEST_PARAMETER_BY_ID)
+  public ResponseEntity<Object> deleteTestParameter(@PathVariable Long id) {
+    if (testParameterService.isTestParameterExist(id)) {
+      testParameterService.deleteTestParameter(id);;
```
[the remark this answers is not in the document]
Comment (ramz7) [reply tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/TestParameterController.java]: yes.it runs, but i will correct

**Pull request merged**